patches/server/0383-Improve-fix-EntityTargetLivingEntityEvent.patch

```diff
@@ -33,13 +33,12 @@ index 5aeef564cdaabeed88a52635e56073cca3a9d1f1..fe635e46569c67dac1d3581ee930d1bf
                          if (event.isCancelled()) {
                              return false;
                          }
--                        if (event.getTarget() == null) {
--                            memoryaccessor.erase();
+-                        if (event.getTarget() != null) {
+-                            entityinsentient.getBrain().setMemory(MemoryModuleType.ATTACK_TARGET, ((CraftLivingEntity) event.getTarget()).getHandle());
 -                            return true;
 -                        }
--                        entityliving = ((CraftLivingEntity) event.getTarget()).getHandle();
 +                        // if (event.getTarget() == null) { // Paper - this is wrong, you are skipping the forgetCallback
-+                        //     memoryaccessor.erase();
++                        //     entityinsentient.getBrain().setMemory(MemoryModuleType.ATTACK_TARGET, ((CraftLivingEntity) event.getTarget()).getHandle());
```

Comment:
I think upstream fixed this completely [here](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/e77bb26bb7c920e9106bf4d2da58dc6b450f4d00#nms-patches%2Fnet%2Fminecraft%2Fworld%2Fentity%2Fai%2Fbehavior%2FBehaviorAttackTargetForget.patch). Our diff can probably be reverted then.